### ISSUE TITLE
Reproduce the experiment in JAFAR's setting

### ISSUE THREAD

**Issue by dyccyber** [valeoai/NAF#6]
I found that using NAF checkpoint evaluation on the jafar codebase yielded poor results.(dinov2-s/dinov3-b）
I'm wondering if it's a checkpoint issue.
The experiment was conducted on a single A100 using the latest version of natten kernelsize=6 due to the image size
using dinov3-b
bilnear
Voc mlou acc  86.95  96.98
naf
Voc mlou 77.82 acc 94.14

**Comment by LoickCh** [comment]
Hi,
Thank you for your interest in our work! 
I have relaunched every results and I have the corresponding log files so I am pretty confident about the scores.

Maybe you can first give me the command you had executed in order to see if I see something wrong and to launch the command line on my side. And then we can investigate together !

**Comment by dyccyber** [comment]
eval log:
Wed Feb 11 17:09:28 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.54.14              Driver Version: 550.54.14      CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A800-SXM4-80GB          On  |   00000000:55:00.0 Off |                    0 |
| N/A   31C    P0             57W /  400W |       0MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+

==================================================
Starting at 2026-02-11 17:10:01
Configuration:
model:
  _target_: src.upsampler.NAF
  dim: 256
  heads_attn: 4
  heads_rope: 4
  kernel_size: 6
  feature_dim: ${get_feature:${backbone.name}}
  use_semencoder: true
  use_encoder: true
  img_layers: 2
  rope_rescale: 2
  name: naf
backbone:
  _target_: src.backbone.PretrainedViTWrapper
  name: vit_base_patch16_dinov3.lvd1689m
  norm: true
optimizer:
  _target_: torch.optim.AdamW
  lr: 0.0005
  betas:
  - 0.9
  - 0.999
  weight_decay: 1.0e-05
loss:
  _target_: src.loss.Loss
  loss_type: cosine_mse
  dim: ${model.feature_dim}
train_dataset:
  _target_: dataloader.ImageDataset
  root: ${dataroot}/train
  root_cache: null
val_dataset:
  _target_: dataloader.ImageDataset
  root: ${dataroot}/val_test
  root_cache: null
train_dataloader:
  _target_: torch.utils.data.DataLoader
  dataset: ${dataset.train}
  batch_size: 2
  shuffle: true
  worker_init_fn:
    _target_: utils.training.seed_worker
  num_workers: 8
  pin_memory: true
val_dataloader:
  _target_: torch.utils.data.DataLoader
  dataset: ${dataset.val}
  batch_size: 1
  shuffle: true
  worker_init_fn:
    _target_: utils.training.seed_worker
  num_workers: 8
  pin_memory: true
img_size: 448
epochs: 1
max_steps: 100000
project_root: ${hydra:runtime.cwd}
dataroot: /.....
bfloat16: true
ratio: random
sanity: false
dataset_evaluation:
  _target_: evaluation.dataset.voc.VOCDataset
  root: /..../seg/
  split: train
  year: '2012'
  download: false
  include_labels: true
  num_classes: 21
  tag: voc
eval:
  task: seg
  model_ckpt: /...../naf.pth
num_epochs: 20
target_size: ${img_size}
metrics:
  seg:
    num_classes: ${dataset_evaluation.num_classes}
  depth:
    min_depth: 0.001
    max_depth: 10

Using device: cuda

Processing seg task:

Image size: 448
[2026-02-11 17:10:02,613][timm.models._builder][INFO] - Loading pretrained weights from Hugging Face hub (timm/vit_base_patch16_dinov3.lvd1689m)
[2026-02-11 17:10:03,339][timm.models._hub][INFO] - [timm/vit_base_patch16_dinov3.lvd1689m] Safe alternative available for 'pytorch_model.bin' (as 'model.safetensors'). Loading weights using safetensors.
Loaded model from checkpoint: 
/dataroot/naf.pth
Train Dataset size: 1464
Val Dataset size: 1449
Training classifier... ./checkpoints/seg/naf.pth not found

[1;36mNumber of optimized parameters: [0m[1;36m16[0m[1;36m,[0m[1;36m149[0m
Training for 20 epochs
..........
[1;32mResults: [0m[1;32m{[0m[1;32m'accuracy'[0m[1;32m: [0m[1;32m0.9414492249488831[0m[1;32m, [0m[1;32m'iou'[0m[1;32m: [0m[1;32m0.7782996892929077[0m[1;32m}[0m
[1;32mTraining completed. Model saved at: .[0m[1;32m/checkpoints/seg/[0m[1;32mnaf.pth[0m

**Comment by dyccyber** [comment]
command :
python evaluation/train_probes.py eval.task=seg dataset_evaluation=voc \
  backbone.name=vit_base_patch16_dinov3.lvd1689m \
  eval.model_ckpt=/..../naf.pth \
  hydra.run.dir=evaluation/unsupervised/voc/vit_base_patch16_dinov3.lvd1689m

Thank you very much for your reply！

**Comment by LoickCh** [comment]
Hi,

I do not see any issue with your command line, but there is a weird formatting of your log file, can you send it to me to loick.chambon@valeo.com? 

Thanks in advance

**Comment by LoickCh** [comment]
Hi,
I have relaunched the command I do not see any error and still do not understand why you have such results. Waiting for the full log !

**Comment by dyccyber** [comment]
Sorry for the late reply. The email has been sent . I am also confused by these results. Thank you very much.!

**Comment by dyccyber** [comment]
I found that using Torch Hub for automatic downloads solved the problem.

However, this issue reappeared when evaluating my own code, possibly due to some hidden code bug. Anyway, my problem is solved.

By the way, there's a potential problem in your code. in model's forward code:
 out = self.upsampler(queries, keys, values, image)
image should be removed

**Comment by LoickCh** [comment]
Ok, let me know if you still have some troubles and do not hesitate to open a new issue.
Thanks again for your interest in our work !
